tests/middleware/test_message_state.py

```diff
@@ -1,30 +1,43 @@
 import time
+from datetime import datetime, timedelta
 
 import pytest
 
 import remoulade
 from remoulade.cancel import Cancel
 from remoulade.middleware import Middleware, SkipMessage
 from remoulade.state.backend import State, StateNamesEnum
-from remoulade.state.middleware import MessageState
+from remoulade.state.middleware import MessageState, State
 
 
 class TestMessageState:
     """Class to test the middleware
     MessageState
     """
 
-    def test_pending_state_message(self, stub_broker, state_middleware, do_work):
+    def test_pending_state_message(self, stub_broker, state_middleware, do_work, frozen_datetime):
         msg = do_work.send()
-        assert state_middleware.backend.get_state(msg.message_id).name == StateNamesEnum.Pending
+        state = state_middleware.backend.get_state(msg.message_id)
+        assert state.message_id == msg.message_id
+        assert state.name == StateNamesEnum.Pending
+        # do work takes one second from [0, 1]s then enqueue should be in second [1, 2]s
+        assert state.enqueued_datetime.isoformat() == "2020-02-03T00:00:00+00:00"
 
-    def test_success_state_message(self, stub_broker, stub_worker, state_middleware, do_work):
+    def test_success_state_message(self, stub_broker, stub_worker, state_middleware, do_work, frozen_datetime):
         msg = do_work.send()
         stub_broker.join(do_work.queue_name)
         stub_worker.join()
-        assert state_middleware.backend.get_state(msg.message_id).name == StateNamesEnum.Success
-
-    def test_started_state_message(self, stub_broker, state_middleware, stub_worker):
+        # Change manually the end time
```

Comment:
You should not manually change the datetime in the state. But using frozen_datetime.tick(delta=15), you can say the time advanced of 15 seconds at this point.

* call `frozen_datetime.tick(delta=15)` just after do_work.send() 
* you define an actor do worck which call `frozen_datetime.tick(delta=15)`

Then verify enqueued_datetime, start_datetime and end_datetime all have 15s between them.